pkg/storage/fs/s3ng/blobstore/blobstore.go

```diff
@@ -62,8 +77,31 @@ func New(endpoint, region, bucket, accessKey, secretKey string) (*Blobstore, err
 	}, nil
 }
 
+// Read implements the read function of the PrometheusAwareReader
+func (p *PrometheusAwareReader) Read(b []byte) (n int, err error) {
+	n, err = p.r.Read(b)
+	p.m.WithLabelValues().Add(float64(n))
+	return
+}
+
+// Read implements the read function of the PrometheusAwareReadCloser
+func (p *PrometheusAwareReadCloser) Read(b []byte) (n int, err error) {
+	n, err = p.r.Read(b)
+	p.m.WithLabelValues().Add(float64(n))
+	return
+}
+
+// Close implements the close function of the PrometheusAwareReadCloser
+func (p *PrometheusAwareReadCloser) Close() error {
+	return p.r.Close()
+}
+
 // Upload stores some data in the blobstore under the given key
 func (bs *Blobstore) Upload(node *node.Node, reader io.Reader) error {
+	reader = &PrometheusAwareReader{
+		r: reader,
+		m: metrics.Tx,
+	}
```

Comment:
why can't we use the `PrometheusAwareReadCloser`? A `ReadCloser` is also a `Reader`?

Reply:
Because it will require lots of changes, which will add cause lots of refactoring when we merge `edge` into our branch again. I tried to avoid this.

Reply:
No, I mean just here in this line. Why don't we do 
```suggestion
	reader = &PrometheusAwareReadCloser{
		r: reader,
		m: metrics.Tx,
	}
```
instead? Then `PrometheusAwareReader` would be unused and could be removed...